azurerm/internal/services/apimanagement/api_management_api_diagnostic_resource.go

```diff
@@ -158,3 +310,48 @@ func resourceArmApiManagementApiDiagnosticDelete(d *schema.ResourceData, meta in
 
 	return nil
 }
+
+func expandApiManagementApiDiagnosticHTTPMessageDiagnostic(input []interface{}) *apimanagement.HTTPMessageDiagnostic {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	result := &apimanagement.HTTPMessageDiagnostic{
+		Body: &apimanagement.BodyDiagnosticSettings{
+			Bytes: utils.Int32(v["body_bytes"].(int32)),
+		},
+	}
```

Comment:
Might be worth checking the `body_bytes` key since it's optional?

```suggestion
	result := &apimanagement.HTTPMessageDiagnostic{
		Body: &apimanagement.BodyDiagnosticSettings{},
	}

	if bodyBytes, ok := v["body_bytes"]; ok {
		result.Body.Bytes = utils.Int32(bodyBytes.(int32)),
	}

```